### PR TITLE
chore: Fully remove `eth_sign`

### PR DESCRIPTION
**Pull request by adonesky1**
## Explanation
Months ago, because of phishing risk, we disabled the `eth_sign` API method by default (users could manually enable it with a preference toggle). Now because of additional risk associated with [potentially malicious EIP-3074 invokers](https://ethereum-magicians.org/t/eip-3074-is-unsafe-unnecessary-puts-user-funds-at-risk-while-fragmenting-ux-liquidity-and-the-wallet-stack/19662) we are fully removing support for this method.

This PR introduces the changes to `@metamask/signature-controller` and `@metamask/preferences-controller` from https://github.com/MetaMask/core/pull/4319 which remove `eth_sign` related infrastructure. 

Additionally it removes all instances of `eth_sign` components and references from the extension codebase.

## References

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/2371

## **Manual testing steps**

1. Go to the [e2e test dapp](https://metamask.github.io/test-dapp/)
2. Connect the wallet
3. Scroll down to the `Eth Sign` card (https://metamask.github.io/test-dapp/#ethSign)
4. Click `Sign`
5. You should see `Error: The method "eth_sign" does not exist / is not available.`

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by adonesky1** [comment]
@metamaskbot update-policies

**Comment by adonesky1** [comment]
@metamask-bot policies-update

**Comment by adonesky1** [comment]
@metamaskbot update-policies

File: ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js [hunk 1]
```diff
@@ -1,426 +0,0 @@
-import React, { Component } from 'react';
```
Comment (digiwand) [inline ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js:1]: I think we'll still need SignatureRequestOriginal to support `eth_signTypedData_v1` aka `eth_signTypedData`. We mentioned it's [deprecated](https://docs.metamask.io/wallet/concepts/signing-methods/#deprecated-signing-methods), but we're currently still supporting it

we can deprecate this one when we remove the "Improve signature requests" setting

**Comment by adonesky1** [comment]
@metamaskbot update-policies

File: app/scripts/controllers/mmi-controller.ts [hunk 2]
```diff
@@ -790,6 +790,13 @@ export default class MMIController extends EventEmitter {
     const isCustodial = Boolean(accountDetails);
     const updatedMsgParams = { ...msgParams, deferSetAsSigned: isCustodial };
 
+    if (
+      !req.method.includes('eth_signTypedData') &&
+      !req.method.includes('personal_sign')
```
Comment (jiexi) [inline app/scripts/controllers/mmi-controller.ts:795]: personal_sign can probably be a `===` check, but tbh i don't love the `req.method.includes('eth_signTypedData')` here in general haha

File: app/scripts/metamask-controller.js [hunk 3]
```diff
@@ -1799,9 +1798,7 @@ export default class MetamaskController extends EventEmitter {
           `${this.loggingController.name}:add`,
         ],
       }),
-      isEthSignEnabled: () =>
-        this.preferencesController.store.getState()
-          ?.disabledRpcMethodPreferences?.eth_sign,
+      isEthSignEnabled: false,
```
Comment (jiexi) [inline app/scripts/metamask-controller.js:1801]: just checking that this can be a constant and not a function.

Doesn't seem like it's used anywhere at all in the SignatureController though

https://github.com/MetaMask/core/blob/main/packages/signature-controller/src/SignatureController.ts#L114

**Review by jiexi — DISMISSED** [review]

File: app/scripts/metamask-controller.js (hunk 3 above)
Comment (adonesky1) [reply app/scripts/metamask-controller.js:1801]: ooof yeah should have been removed in the SignatureController

**Review by jiexi — DISMISSED** [review]

File: app/scripts/metamask-controller.js [hunk 4]
```diff
@@ -1799,9 +1798,8 @@ export default class MetamaskController extends EventEmitter {
           `${this.loggingController.name}:add`,
         ],
       }),
-      isEthSignEnabled: () =>
-        this.preferencesController.store.getState()
-          ?.disabledRpcMethodPreferences?.eth_sign,
+      // TODO remove this option
+      isEthSignEnabled: () => false,
```
Comment (FrederikBolding) [inline app/scripts/metamask-controller.js:1802]: Can we remove this now?

File: ui/pages/confirmations/components/signature-request-original/signature-request-original.test.js [hunk 5]
```diff
@@ -63,10 +27,21 @@ const props = {
   txData: {
     msgParams: {
       from: address,
-      data: MOCK_SIGN_DATA,
+      data: [
```
Comment (FrederikBolding) [inline ui/pages/confirmations/components/signature-request-original/signature-request-original.test.js:30]: Why the change in test data here? IIRC the data is stringified 🤔

File: ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js [hunk 6]
```diff
@@ -313,10 +311,6 @@ export default class SignatureRequestOriginal extends Component {
           history.push(mostRecentOverviewPage);
         }}
         onSubmit={async () => {
-          if (txData.type === MESSAGE_TYPE.ETH_SIGN) {
-            return this.setState({ showSignatureRequestWarning: true });
```
Comment (FrederikBolding) [inline ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js:317]: Can't we delete the state and component for `showSignatureRequestWarning` too?

File: app/scripts/controllers/mmi-controller.ts [hunk 7]
```diff
@@ -790,23 +790,29 @@ export default class MMIController extends EventEmitter {
...
       return await this.signatureController.newUnsignedPersonalMessage(
         updatedMsgParams as PersonalMessageParams,
         req as OriginalRequest,
       );
     }
-    return await this.signatureController.newUnsignedMessage(
-      updatedMsgParams as PersonalMessageParams,
-      req as OriginalRequest,
-    );
+    return undefined;
```
Comment (FrederikBolding) [inline app/scripts/controllers/mmi-controller.ts:815]: Nit: Perhaps just throwing here is simpler?

File: app/scripts/metamask-controller.js (hunk 4 above)
Comment (adonesky1) [reply app/scripts/metamask-controller.js:1802]: Yes. Not sure why but I thought I'd forgotten to remove it as an option so was going to loop back to it but it can be remove removed now.

File: ui/pages/confirmations/components/signature-request-original/signature-request-original.test.js (hunk 5 above)
Comment (adonesky1) [reply ui/pages/confirmations/components/signature-request-original/signature-request-original.test.js:30]: So I didn't change the way the data was formatted in the actual component for `eth_signTypedData` see [here](https://github.com/MetaMask/metamask-extension/pull/24756/files/2219e0ddbe92d95876d9bd3897462f6962389f5d#diff-0b1e3dca811d174401e32d4e02e4f70ebf269526c26c58f788cb7c3f0ee28ce0R132)... I agree this doesn't seem right. But as you can see [here](https://github.com/MetaMask/metamask-extension/blob/7a1191b33f1e5fac284fb7fb9ac1fa61102b0093/ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js#L224), this data is expected to be mappable. So what I've done here is preserve the existing behavior but make the test pass because of expectations embedded in the component that I didn't necessarily want to dig further into because of 👇 

A few things. This component is [technically deprecated](https://consensys.slack.com/archives/C03ETQA9EPK/p1721773925209699?thread_ts=1721768575.997869&cid=C03ETQA9EPK) and slated for removal shortly. I'm not clear on the original API method signature for `eth_signTypedData` (v1) since EIP-712 doesn't have a changelog...

File: ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js (hunk 6 above)
Comment (adonesky1) [reply ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js:317]: https://github.com/MetaMask/metamask-extension/pull/24756/commits/343ee961208c68f42397d6591f73f14d2c74891e

File: app/scripts/metamask-controller.js (hunk 4 above)
Comment (adonesky1) [reply app/scripts/metamask-controller.js]: https://github.com/MetaMask/metamask-extension/pull/24756/commits/343ee961208c68f42397d6591f73f14d2c74891e

**Review by gantunesr — DISMISSED** [review]
Approved for team-accounts.

Just reviewed the file `snap-keyring/metrics.test.ts`

**Review by FrederikBolding — DISMISSED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by gantunesr — APPROVED** [review]
Approved for team-accounts.

Just reviewed the file snap-keyring/metrics.test.ts

**Review by sleepytanya — APPROVED** [review]

File: app/scripts/controllers/mmi-controller.ts [hunk 8]
```diff
@@ -790,23 +790,25 @@ export default class MMIController extends EventEmitter {
...
         { parseJsonData: false },
       );
-    } else if (req.method.includes('personal_sign')) {
+    } else if (req.method === 'personal_sign') {
       return await this.signatureController.newUnsignedPersonalMessage(
         updatedMsgParams as PersonalMessageParams,
         req as OriginalRequest,
       );
     }
-    return await this.signatureController.newUnsignedMessage(
```
Comment (albertolive) [inline app/scripts/controllers/mmi-controller.ts:806]: @adonesky1, the only thing I'm afraid of in this PR is the removal of this `newUnsignedMessage` for the custodian accounts. Let me talk with the team first. cc @zone-live @shane-t

Comment (shane-t) [reply app/scripts/controllers/mmi-controller.ts:806]: I'm not concerned about it, because it willy only be true if the method is `eth_sign` based on the other conditions in the block.

**Review by albertolive — APPROVED** [review]
